client/src/components/ViewContact.jsx

```diff
@@ -2,66 +2,45 @@ import Box from "@mui/material/Box";
 import CircularProgress from "@mui/material/CircularProgress";
 import Snackbar from "@mui/material/Snackbar";
 import Alert from "@mui/material/Alert";
-import { useEffect, useState } from "react";
+import { useEffect, useState, useRef } from "react";
 import EditContactForm from "./EditContactForm";
 import IndividualContact from "./IndividualContact";
 import { fetchContacts } from "../fetchContacts";
 
 const ViewContact = ({ selectedContactId, setContacts }) => {
   const [editButtonClicked, setEditButtonClicked] = useState(false);
-  const [openSnackbar, setOpenSnackbar] = useState(false);
+
   const [snackbarMessage, setSnackbarMessage] = useState("");
+
   const [contactData, setContactData] = useState({
-    newName: "",
-    newEmail: "",
-    newPhone: "",
-    newNotes: "",
-    originalData: {
-      name: "",
-      email: "",
-      phone: "",
-      notes: "",
-    },
+    name: "",
+    email: "",
+    phone: "",
+    notes: "",
   });
+  const originalContactDetails = useRef(null);
```

Comment:
This is how I save the original contact details. Use cmd+F to see where this ref variable is used.